src/accountsdb/db.zig

```diff
@@ -3034,60 +2962,83 @@ pub fn writeSnapshotTarWithFields(
     archive_writer: anytype,
```

Comment:
wdyt about moving this + snapgen logic to `snapshots.zig`? feels like `writeSnapshotTarWithFields` should also be in snapgen too -- would also prefer to be more explicit and name it `snapshot_generation`

Reply:
Actually, I think it would make sense for `writeSnapshotTarWithFields` to remain in accountsdb, because it directly accepts as input accountsdb's file map field. In my mind accountsdb should use snapshots.zig, but we should avoid snapshots.zig using accountsdb if possible - cleaner dependency edge that way.

Reply:
a44491fb7e59ed34f0e02a1f88f72c2c91b0470f lmk wyt